### DISCUSSION TITLE
Can I use "next" storage with Expo?

### DISCUSSION THREAD

**Discussion by isocra** [react-native-async-storage/async-storage#654] [Q&A]
## You want to:  
Ask if it's possible to use the `next` storage implementation with Expo managed flow to build a React Native app for engineers out in the field.

## Details:
I'm using Expo managed flow and Expo bundles, well makes easily available, this AsyncStorage plugin. I'm also using an Apollo offline cache to manage when the device is offline. But I'm getting reports of `database or disk is full` errors.

So is there anything I can put in the app.json or elsewhere so that AsyncStorage uses the `next storage implementation` and does that fix the space issue? I don't get any errors from iOS devices, only Android

**Comment by krizzu** [comment]
Hey @isocra 

Unfortunately, no, it's not possible. `database or disk is full` is interesting one - do you see this on different Android devices or just one?

**Reply by isocra** [reply]
It happens on lots of our engineers' Android devices. According to the docs, AsyncStorage is only 6Mb. I think it's being used by our Apollo offline cache and that's sometimes running out of space. It isn't a problem on iOS as AsyncStorage doesn't have a size limit. As far as I can see, if we could use the `next` storage implementation for Android (which is as simple as setting `AsyncStorage_useNextStorage=true` in gradle.properties see [here](AsyncStorage_useNextStorage=true)), then the problem would go away.

**Reply by krizzu** [reply]
This flag works only for non-Expo managed projects (or ejected). Expo does not use the Async Storage you see here, but Async Storage from `react-native` directly. We have "expo support" that allows for expo users to install this library and not crash due to missing native code.

I guess you could try experiment with your project - clone it again, eject expo, use next storage flag and see if it works for you. I'd also check how big the data is on iOS and check if it's indeed exceeding 6MB.
